### ISSUE TITLE
<img ng-src="{{url()}}">  Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Millen31** [Urigo/angular-meteor#436]
With Angular-Meteor the function will be called 8 times. I just get 2 "yeah"s with pure angular. In different scenarios I got a "Maximum call stack size exceeded"
```
<img ng-src="{{url()}}">

$scope.url = function () {
        console.log("yeah");
        return "http://images.tagseoblog.de/bilder/bild-foto/bild.jpg";
    }
```

**Comment by MilosStanic** [comment]
+1, getting the same error in similar situation, i.e. when calling a controller function to resolve image url.

**Comment by mxab** [comment]
I think the problem here is that angular is not able to check when this needs to be called similar to the stateful filters and therefore evaluates your expression during every digestion.

Is it possible to realise this within a watch expression?
```
<img ng-src="{{url}}">
...
$scope.$watch("something", function(){
$scope.url = calculateUrl(...)
})
```
Just a wild guess

**Comment by pythman** [comment]
@Millen31  I tried your code and it works fine with no errors

**Comment by Urigo** [comment]
@Millen31 can you share this code in a repo that I can run?
or better, in a failing test on the tests folder?
thanks

**Comment by Millen31** [comment]
@Urigo Here is my repo. I just added the CollectionFS-Example.

https://github.com/Millen31/angular_meteor_test

**Comment by netanelgilad** [comment]
Ok, now I see what is going on here. The problem is that we use `diffObjectRemovals` on a `FS.File` object which is a unique object. It is actually a wrapper around the files in the CollectionFS collection. So that wrapper creates an object with references to itself in nested properties. That causes `diffObjectRemovals` to never break out of the recursion since it doesn't handle circular references.

So two ways I think to go about this:
1. add circular reference handling to `diffObjectRemovals`. That may solve the problem but this is a very special use case, since I don't think regular Mongo documents can contain circular reference.
2. Find a different way to handle `FS.File` object (since I think deep copying a special object like that may break stuff since it has reference to actual File objects, but I'm not sure about that).

@Urigo @Millen31 @MilosStanic @mxab thoughts?

**Comment by dotansimha** [comment]
Hi @Millen31  @MilosStanic  @mxab ,
This PR fixes this issue by adding $meteor.collectionFS for handling this kind of collections.
https://github.com/Urigo/angular-meteor/pull/467

**Comment by Urigo** [comment]
Solved in 0.9.2

**Comment by darthrayne** [comment]
yes, should be `$scope.$meteorSubscribe('images').then(function () {
            $scope.images = $meteor.collectionFS(Images, false, Images);
            console.log([$scope.images]);
        });`
